### ISSUE TITLE
Update check fails

### ISSUE THREAD

**Issue by bontchev** [jimradford/superputty#689]
A couple of days ago the update check stopped working - it just produces an error message, saying that an error has occurred.

I don't know what is causing it, but if you are checking for new versions on GitHub, please note that GitHub recently disabled support for insecure HTTPS protocols, like TLS 1.0. Maybe this is what is causing the problem.

**Comment by Denisnow99** [comment]
I am getting same message Updates not working

**Comment by agallo** [comment]
[Github's announcement](https://githubengineering.com/crypto-removal-notice/)

TLS error:
```
TLSv1 Record Layer: Alert (Level: Fatal, Description: Protocol Version)
    Content Type: Alert (21)
    Version: TLS 1.0 (0x0301)
    Length: 2
    Alert Message
        Level: Fatal (2)
        Description: Protocol Version (70)
```

**Comment by jimradford** [comment]
Thanks for the reports, I'll look into this deeper as we don't specify the protocol (other than it be secure https) being used for update checks.

**Comment by overproof876** [comment]
Just confirming i'm also getting the update error

**Comment by bontchev** [comment]
@jimradford, the problem might be in the HTTPS library you use for the requests. Unfortunately, I don't know C#, so I can't help, but check if there is an update for that library and what parameters you are using.

**Comment by jimradford** [comment]
Really shouldn't be, its the built in .NET libraries. I'll try and throw
it on a Debugger this afternoon and see specifically what the problem
is. It should be pretty simple actually. The bummer will be the update
checker is broken so people will have to manually check until they
download a new version. 

Jim

-- 
Jim Radford <jradford@npl.com>
http://www.jimradford.com 

On 2018-02-27 10:13, Vesselin Bontchev wrote:

> @jimradford [1], the problem might be in the HTTPS library you use for the requests. Unfortunately, I don't know C#, so I can't help, but check if there is an update for that library and what parameters you are using. 
> 
> --
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub [2], or mute the thread [3].
 

Links:
------
[1] https://github.com/jimradford
[2]
https://github.com/jimradford/superputty/issues/689#issuecomment-368973178
[3]

**Comment by bontchev** [comment]
As I said, I don't really know C#, so I can't be very helpful about how to fix it. Maybe the ``WebRequest`` class uses insecure methods by default, or something. Try putting

```
ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
```

at the beginning of the ``MakeRequest`` method.

**Comment by jimradford** [comment]
I recommend disabling the update check (Tools -> Options -> Advanced -> Uncheck Automatically check for updates) since the functionality is broken

or edit the config file SuperPuTTY.settings:

```
setting name="AutoUpdateCheck" serializeAs="String">
        <value>True</value>

to

setting name="AutoUpdateCheck" serializeAs="String">
        <value>False</value>

```
 since its now broken functionality. We'll get an update out soon that addresses the issue.

**Comment by jimradford** [comment]
Ok, so I did some additional checking and the problem may have been caused by Github changing their support for older ssl protocols and requiring all connections to use tls 1.2.

Currently SuperPutty is set to use the .NET framework 3.5, this version of the framework does not support TLS 1.2 without installing a patch. Even if that patch were installed it would still require code changes to support the change. See this article for additional information: https://support.microsoft.com/en-us/help/3154518/support-for-tls-system-default-versions-included-in-the-net-framework 

I don't think that solution is viable, it creates more work for the end user of the application and some might not be that technical.

Option 2 would be to update the application to support the .NET 4.5 framework which does support tls1.2, code changes and a release will still be required but really that only means downloading and installing a new version of SuperPuTTY.

**Comment by bontchev** [comment]
I think Google disabled only TLS 1.0, which is insecure. TLS 1.1 should still be supported. I don't know if .NET 3.5 supports it, though.

**Comment by jimradford** [comment]
.NET 3.5 does not support TLS 1.2, .NET 4.0 does with a code change (tls
version is not in the Enum so it has to be cast), .NET 4.5 and newer do.

>

**Comment by bontchev** [comment]
Yes, I got it that .NET 3.5 does not support TLS 1.2 while .NET 4.5 does. I was wondering whether .NET 3.5 supports TLS 1.1 (not 1.2); I think that GitHub disabled only TLS 1.0, so TLS 1.1 will probably still work. But, yes, probably the best solution is to rewrite the program to use .NET 4.5 and TLS 1.2.

**Comment by jimradford** [comment]
@bontchev If you wouldn't mind, can you grab and test a nightly build from here: https://ci.appveyor.com/api/buildjobs/rk8xja6427vy6y33/artifacts/SuperPuttyInstaller%2Fbin%2FDebug%2FSuperPuttySetup.msi and confirm it works. This is a version with the update checker fixed but requiring .NET 4.5. iirc tls 1.1 is not supported in .NET 3.5 either.

**Comment by gurnec** [comment]
The update check w/the nightly worked for me (Win10 fall creators update), thanks.

**Comment by bontchev** [comment]
I cannot confirm. After installing the package, my executable hasn't changed - it is still version 1.4.0.8 and still an error occurs when it tries to check for updates. My environment is Windows 7 Pro, 64-bit

I could try to uninstall the old version first - but will I lose my settings and saved sites?

**Comment by jimradford** [comment]
Hi, 

You should be able to safely uninstall the old version and install the
test version. Or you can grab just the exe and replace the installed one
with it. 

>

**Comment by gurnec** [comment]
@bontchev My executable is also claiming to be 1.4.0.8, so the version number apparently wasn't updated. Maybe the installer failed for some reason? Did you have SuperPutty open during the install? Have you rebooted since the install? (FYI I uninstalled the old version first; I didn't take note if the settings were preserved, but the saved sites definitely were.)

The SHA-1 of the new executable is a76e18e2c2feb9cfd8757eaf57dda2966d15017e, does yours match?

**Comment by bontchev** [comment]
@jimradford, how can I "grab just the exe"?

@gurnec, No, the program wasn't running when I tried to update, of course. The installer didn't fail, in the sense that there was no error - but my executable definitely wasn't updated. Its compile date is old (2015), and the SHA1 is ``a76e18e2c2feb9cfd8757eaf57dda2966d15017e``.

**Comment by jimradford** [comment]
You can grab it from the CI build server artifacts output:
https://ci.appveyor.com/project/jimradford/superputty/branch/master/artifacts


>

**Comment by bontchev** [comment]
OK, if I replace the executable from there, the update check works correctly (no errors).

**Comment by jimradford** [comment]
@bontchev  it will be included when we do another release. For now people can grab the EXE from the CI Server or reinstall with the installer from the CI server. 

Thanks for the effort testing this.
Jim

**Comment by jimradford** [comment]
Hi Kevin, 

Nothing related to passing command parameters to putty was touched to
fix the update checker, however it may have changed depending on which
version you were using before you updated. That area of the application
code hasn't changed in many years, so I would be surprised if it was a
change we did. 

I do however recommend using key authentication instead of the less
secure passing passwords in plaintext (-pw) method for authentication. 

Jim 

On 2018-04-07 11:22, kevinmarchant wrote:

> Hi Jim, 
> 
> I just installed the CI server version (did an uninstall and reinstall to make sure the revised version was installed). 
> 
> I'm no longer getting the update check error (yeah!), but 
> 
> On some of my Putty configs to automate login I pass the password using "-pw pw_in_clear" in Extra PuTTy arguments. This doesn't seem to be working in the revised version - it's passing the username, but stopping to prompt for the password. Is it failing to append Extra PuTty arguments? Or have I done something stupid ? 
> 
> --
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/jimradford/superputty/issues/689#issuecomment-379489298
[2]

**Comment by kevinmarchant** [comment]
Thanks Jim, I had upgraded from 1.4.0.8.  I reverted to the old 1.4.0.8 build, and set autoupdate off in the .settings file to avoid the warning .  Behaviour is back to where I wanted it with -pw xxxx being honoured after installing un-fixed version.

 

I agree with the security point, but I only use it in that way for a couple of NAS boxes at home so couldn’t be bothered with certificate-based.

 

Not sure why “-pw” it works with 1.4.0.8 and not 1.4.0.8 (with update fix) for an SSH connection over default port 22.

 

Anyway, thanks for the reply – useful program !

 

Kevin

**Comment by PieterMeyer** [comment]
Kevin,
I also found this and it seems that the security setting to check plain text password was not previously not enforced.
Check that the setting is selected under Option --> Gui tab --> Security --> Allow plain text passwords on putty command line.
Pieter.
